command/acl_auth_method_create.go

```diff
@@ -0,0 +1,174 @@
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/nomad/api"
+	"github.com/mitchellh/cli"
+	"github.com/posener/complete"
+	"golang.org/x/exp/slices"
+)
+
+// Ensure ACLRoleCreateCommand satisfies the cli.Command interface.
+var _ cli.Command = &ACLAuthMethodCreateCommand{}
+
+// ACLAuthMethodCreateCommand implements cli.Command.
+type ACLAuthMethodCreateCommand struct {
+	Meta
+
+	name          string
+	methodType    string
+	tokenLocality string
+	maxTokenTTL   time.Duration
+	isDefault     bool
+	config        string
+}
+
+// Help satisfies the cli.Command Help function.
+func (a *ACLAuthMethodCreateCommand) Help() string {
+	helpText := `
+Usage: nomad acl auth-method create [options]
+
+  Create is used to create new ACL auth methods. Use requires a management token.
+
+General Options:
+
+  ` + generalOptionsUsage(usageOptsDefault|usageOptsNoNamespace) + `
+
+ACL Auth Method Create Options:
+
+  -name
+    Sets the human readable name for the ACL auth method. The name must be
+    between 1-128 characters and is a required parameter.
+
+  -type
+    Sets the type of the auth method. Currently the only supported type is
+    'OIDC'.
+
+  -max-token-ttl
+    Sets the duration of time all tokens created by this auth method should be
+    valid for.
+
+  -token-locality
+    Defines the kind of token that this auth method should produce. This can be
+    either 'local' or 'global'. If empty the value of 'local' is assumed.
+
+  -default
+    Specifies whether this auth method should be treated as a default one in
+    case no auth method is explicitly specified for a login command.
+
+  -config
+    Auth method configuration in JSON format.
+`
+	return strings.TrimSpace(helpText)
+}
+
+func (a *ACLAuthMethodCreateCommand) AutocompleteFlags() complete.Flags {
+	return mergeAutocompleteFlags(a.Meta.AutocompleteFlags(FlagSetClient),
+		complete.Flags{
+			"-name":           complete.PredictAnything,
+			"-type":           complete.PredictAnything,
+			"-max-token-ttl":  complete.PredictAnything,
+			"-token-locality": complete.PredictAnything,
+			"-default":        complete.PredictAnything,
+			"-config":         complete.PredictNothing,
+		})
+}
+
+func (a *ACLAuthMethodCreateCommand) AutocompleteArgs() complete.Predictor {
+	return complete.PredictNothing
+}
+
+// Synopsis satisfies the cli.Command Synopsis function.
+func (a *ACLAuthMethodCreateCommand) Synopsis() string { return "Create a new ACL auth method" }
+
+// Name returns the name of this command.
+func (a *ACLAuthMethodCreateCommand) Name() string { return "acl auth-method create" }
+
+// Run satisfies the cli.Command Run function.
+func (a *ACLAuthMethodCreateCommand) Run(args []string) int {
+
+	flags := a.Meta.FlagSet(a.Name(), FlagSetClient)
+	flags.Usage = func() { a.Ui.Output(a.Help()) }
+	flags.StringVar(&a.name, "name", "", "")
+	flags.StringVar(&a.methodType, "type", "", "")
+	flags.StringVar(&a.tokenLocality, "token-locality", "local", "")
```

Comment:
This default create an inconsistent UX between the API and CLI. I wonder if we want to move this defaulting into the RPC layer via a canonicalize function and default to empty here, allowing that through the CLI validation? The RPC part can be done as a follow up.

Reply:
Agreed. I'm removing the default here.